x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/eql/action/SubmitAsyncEqlSearchRequest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.eql.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskId;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+/**
+ * A request to track asynchronously the progress of an eql search against one or more indices.
+ *
+ * @see AsyncEqlSearchResponse
+ */
+public class SubmitAsyncEqlSearchRequest extends ActionRequest {
```

Comment:
same here, could/should we share the request?

Reply:
As we discussed yesterday this one goes away once I merge async and non sync method together. Even if we didn't merge them, this method wraps EqlSearchRequest while  SubmitAsyncSearchRequest wraps SearchRequest and has several additional attributes that this one doesn't. So I don't think we could share it at least easily.

Reply:
I guess I meant that we could share the stub like wait for completion and keep alive, but I guess we would have also in this case problems due to multiple inheritance that is not possible

Reply:
Yes, looking into this now. Need to resolve fallout from the last night merge first though.